### ISSUE TITLE
SC1090/1091 could be emitted only if the lack of file is actually going to cause a problem

### ISSUE THREAD

**Issue by anntzer** [koalaman/shellcheck#1325]
#### For bugs
- Rule Id (if any, e.g. SC1000): SC1090/SC1091
- My shellcheck version (`shellcheck --version` or "online"): 0.5.0
- [X] I tried on shellcheck.net and verified that this is still a problem on the latest commit
- [X] It's not reproducible on shellcheck.net, but I think that's because it's an OS, configuration or encoding issue

#### For new checks and feature suggestions
- [X] shellcheck.net (i.e. the latest commit) currently gives no useful warnings about this
- [X] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related

#### Here's a snippet or screenshot that shows the problem:

The docs for SC1090 state that not following sourced files can "potentially result in warnings about unassigned variables and similar."  It would be nice if SC1090 was only emitted if SC is actually about to emit an unassigned variable warning.

Example use case: shell scripts that dynamically create a Python virtualenv in a temporary directory and source-activate it.  This sets some environment variables that are only being read by Python, but won't even be mentioned in the shell script.

```sh
#!/bin/bash
venvpath="$(mktemp -u)"
trap 'rm -rf "$venvpath"' EXIT
python -mvenv "$venvpath"
(
    source "$venvpath/bin/activate"
    python -c 'print("hello world")'
)
```

#### Here's what shellcheck currently says:

SC1090: Can't follow non-constant source. Use a directive to specify location.

#### Here's what I wanted or expected to see:

No warning.

**Comment by RobertKielty** [comment]
Could a possible solution here could be to alter the behaviour of the shellcheck source directive to account for the build/check time expected absence of the dynamically invoked script

# shellcheck source=./possible/path/to/included/script if-not-present=[ignore,emit_warning,emit_error]
